packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -164,7 +158,57 @@ class DriveCommand extends RunCommandBase {
         );
       }
 
-      final LaunchResult result = await appStarter(this);
+      if (isWebPlatform && getBuildInfo().isDebug) {
+        // TODO(angjieli): remove this once running against
+        // target under test_driver in debug mode is supported
+        throwToolExit(
+          'Flutter Driver web does not support running in debug mode.\n'
+          '\n'
+          'Use --profile mode for testing application performance.\n'
+          'Use --release mode for testing correctness (with assertions).'
+        );
+      }
+
+      Uri webUri;
+
+      if (isWebPlatform) {
+        ResidentRunner runner;
+        final FlutterProject flutterProject = FlutterProject.current();
+        final FlutterDevice flutterDevice = await FlutterDevice.create(
+            device,
+            flutterProject: flutterProject,
+            trackWidgetCreation: boolArg('track-widget-creation'),
+            target: targetFile,
+            buildMode: getBuildMode()
+        );
+        runner = webRunnerFactory.createWebRunner(
+          flutterDevice,
+          target: targetFile,
+          flutterProject: flutterProject,
+          ipv6: ipv6,
+          debuggingOptions: DebuggingOptions.enabled(getBuildInfo()),
+          stayResident: false,
+          dartDefines: dartDefines,
+          urlTunneller: null,
+        );
+
+        // Sync completer so the completing agent attaching to the resident doesn't
+        // need to know about analytics.
+        //
+        // Do not add more operations to the future.
+        final Completer<void> appStartedTimeRecorder = Completer<void>.sync();
```

Comment:
Should you be awaiting this so that you're sure runner.uri is not null? Also, make it a regular completer and rename to reflect its current purpose

Reply:
Done.